lib/matplotlib/axes/_axes.py

```diff
@@ -1673,6 +1674,7 @@ def semilogy(self, *args, **kwargs):
             self.cla()
         d = {'basey': kwargs.pop('basey', 10),
              'subsy': kwargs.pop('subsy', None),
+             'nonposy': kwargs.pop('nonposy', 'mask'),
```

Comment:
Because typo.  But actually both should be using "mask" mode if we want to keep backcompatibility (that's what we had before).  Feel free to push either option (but it seems normal to return to the 2.0.1 behavior for now).